### ISSUE TITLE
WARNING:tensorflow:Can save best model only with val_loss available, skipping.

### ISSUE THREAD

**Issue by ManuelPopp** [keras-team/keras#16734]
**System information**.
Windows 8.1.
Python version: 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]
Tensorflow version: 2.9.1
as well as
Linux 4.18.0-305.40.2.el8_4.x86_64.
Python version: 3.8.6 (default, Jan 22 2021, 11:41:28) 
[GCC 8.4.1 20200928 (Red Hat 8.4.1-1)]
Tensorflow version: 2.8.0

**Describe the problem**.
I get the error message in any case where I pass something other than "loss" to the monitor argument. I tried it using `"val_loss"`, as well as all combinations with and without the `"val_"` prefix for the metrics I monitor. I did check history.history to copy the metric names from there.

**Describe the current behavior**.
When including
```
cllbs = [ks.callbacks.ModelCheckpoint(os.path.join("/some/directory/", \
                                              "Epoch.{epoch:02d}.hdf5"), \
                                 monitor = "val_loss", \
                                 mode = monitor_mode, \
                                 save_best_only = True, \
                                 save_freq = 5)]
in
```
```
args_fit = {"epochs" : 100,
            "steps_per_epoch" : np.ceil(N_img/bs),
            "validation_steps" : np.ceil(N_val/bs),
            "callbacks" : cllbs}
hist = model.fit(dataset["train"].map(add_sample_weights),
                         validation_data = dataset["val"],
                         **args_fit)
```
I receive the warning `"WARNING:tensorflow:Can save best model only with val_loss available, skipping."` and the model is not saved at checkpoints.

**Describe the expected behavior**.
I expect the callback to recognize the metric argument and save the model.
The documentation on https://keras.io/api/callbacks/model_checkpoint/ explicitly states "Use "loss" or "val_loss" to monitor the model's total loss."
**Standalone code to reproduce the issue**.

```
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
inputs = keras.Input(shape=(784,), name="digits")
x = layers.Dense(64, activation="relu", name="dense_1")(inputs)
x = layers.Dense(64, activation="relu", name="dense_2")(x)
outputs = layers.Dense(10, activation="softmax", name="predictions")(x)

model = keras.Model(inputs=inputs, outputs=outputs)
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

# Preprocess the data (these are NumPy arrays)
x_train = x_train.reshape(60000, 784).astype("float32") / 255
x_test = x_test.reshape(10000, 784).astype("float32") / 255

y_train = y_train.astype("float32")
y_test = y_test.astype("float32")

# Reserve 10,000 samples for validation
x_val = x_train[-10000:]
y_val = y_train[-10000:]
x_train = x_train[:-10000]
y_train = y_train[:-10000]
model.compile(
    optimizer=keras.optimizers.RMSprop(),  # Optimizer
    # Loss function to minimize
    loss=keras.losses.SparseCategoricalCrossentropy(),
    # List of metrics to monitor
    metrics=[keras.metrics.SparseCategoricalAccuracy()],
)

import tempfile, os
tmp_dir = tempfile.mkdtemp()
monitor_mode = "min"
cllbs = [
    keras.callbacks.ModelCheckpoint(os.path.join(tmp_dir, \
                                              "Epoch.{epoch:02d}.hdf5"), \
                                 monitor = "val_loss", \
                                 mode = monitor_mode, \
                                 save_best_only = True, \
                                 save_freq = 5)
    ]

history = model.fit(
    x_train,
    y_train,
    batch_size=64,
    epochs=2,
    validation_data=(x_val, y_val),
    callbacks = cllbs
)
```
This example was taken from https://www.tensorflow.org/guide/keras/train_and_evaluate and I only added the callbacks part.
Output:
```
Downloading data from https://storage.googleapis.com/tensorflow/tf-keras-datasets/mnist.npz
11490434/11490434 [==============================] - 2s 0us/step
Epoch 1/2
  1/782 [..............................] - ETA: 4:16 - loss: 2.3448 - sparse_categorical_accuracy: 0.0469WARNING:tensorflow:Can save best model only with val_loss available, skipping.
WARNING:tensorflow:Can save best model only with val_loss available, skipping.
  5/782 [..............................] - ETA: 12s - loss: 2.1246 - sparse_categorical_accuracy:
```
... etc pp

**Comment by sushreebarsa** [comment]
@ManuelPopp Thank you for the update!
Could you please confirm if this issue is resolved for you?
Thank you!

**Comment by ManuelPopp** [comment]
Yes, setting save_freq = "epoch" solved the issue.
